pkg/services/dashboards/folder_service.go

```diff
@@ -0,0 +1,246 @@
+package dashboards
+
+import (
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/guardian"
+	"github.com/grafana/grafana/pkg/services/search"
+)
+
+// FolderService service for operating on folders
+type FolderService interface {
+	GetFolders(limit int) ([]*models.Folder, error)
+	GetFolderById(id int64) (*models.Folder, error)
+	GetFolderByUid(uid string) (*models.Folder, error)
+	CreateFolder(cmd *models.CreateFolderCommand) error
+	UpdateFolder(uid string, cmd *models.UpdateFolderCommand) error
+	DeleteFolder(uid string) (*models.Folder, error)
+}
+
+// NewFolderService factory for creating a new folder service
+var NewFolderService = func(orgId int64, user *models.SignedInUser) FolderService {
```

Comment:
I don't think orgId and user should be a part of the constructor. IMO services/repos should not be bound to the request lifecycle. I guess its a matter of taste. I think would be a great code talk about how we want to build things in the backend. 

But it's **not** a blocker. This can be **refactored later**.

Reply:
It was a nice solution when there wasn't any command/queries as method arguments. In one way it's kind of nice that you cannot get a hold of such instance if you don't provide orgId and SignedInUser. Makes it explicit. But in the other hand I would have preferred having queries/commands as arguments for all methods, but it's a bit weird that you may expect to be able to dispatch them with the bus?

I agree that we really should have a discussion regarding this soon!